File: source/com/barrybecker4/common/expression1/Operator.java
// Copyright by Barry G. Becker, 2013. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.common.expression1;

/**
 * The expected binary operators in the text expression.
 * @author Barry Becker
 */
public interface Operator  {

    char getSymbol();

    double operate(double operand1, double operand2);
}

File: scala-source/com/barrybecker4/common/expression/Operator.scala
/* Copyright by Barry G. Becker, 2000-2018. Licensed under MIT License: http://www.opensource.org/licenses/MIT */
package com.barrybecker4.common.expression

/**
  * The expected binary operators in the text expression.
  * @author Barry Becker
  */
abstract class Operator(val symbol: Char) {
  def operate(operand1: Double, operand2: Double): Double
}